core/include/traccc/fitting/kalman_filter/gain_matrix_smoother.hpp

```diff
@@ -110,8 +112,14 @@ struct gain_matrix_smoother {
         cur_state.smoothed().set_vector(smt_vec);
         cur_state.smoothed().set_covariance(smt_cov);
 
-        const matrix_type<D, e_bound_size> H =
-            meas.subs.template projector<D>();
+        matrix_type<D, e_bound_size> H = meas.subs.template projector<D>();
+
+        // Correct sign for line detector
+        if (mask_t::shape::mask_type == detray::mask_types::line) {
```

Comment:
@niermann999 You can see here how I am using `shape_type`

Reply:
What I meant was 
```
using shape_t = typename mask_t::shape;
if constexpr (std::is_same_v<shape_t, detray::line<true>> || std::is_same_v<shape_t, detray::line<false>>) {
}
```
but nvm

Reply:
Hmm I think it is better.